src/components/DrawCore/index.tsx

```diff
@@ -254,1044 +255,1098 @@ const DrawCore = React.forwardRef<
     image?: ImageRequireSource | ImageURISource;
     linearGradient: React.ComponentType<{ colors: any[] } & ViewProps>;
     onSelectionChange?: (selected: boolean) => void;
+    onCancelChange?: (cancel: boolean) => void;
   }
->(({ drawingMode, image, linearGradient, onSelectionChange }, ref) => {
-  const mode = useSharedValue<DrawItemType>('pen');
+>(
+  (
+    { drawingMode, image, linearGradient, onSelectionChange, onCancelChange },
+    ref
+  ) => {
+    const mode = useSharedValue<DrawItemType>('pen');
 
-  const [drawRegion, setDrawRegion] = useState<Size | null>(null);
+    const [drawRegion, setDrawRegion] = useState<Size | null>(null);
 
-  const [originalImageSize, setOriginalImageSize] = useState<Size | null>(null);
+    const [originalImageSize, setOriginalImageSize] =
+      useState<Size | null>(null);
 
-  const [imageSize, setImageSize] = useState<Size | null>(null);
+    const [imageSize, setImageSize] = useState<Size | null>(null);
 
-  const drawContainer = useRef<View>(null);
+    const drawContainer = useRef<View>(null);
 
-  const viewShot = useRef<ViewShot>(null);
+    const viewShot = useRef<ViewShot>(null);
 
-  const [textVal, setTextVal] = useState<string>('');
+    const [textVal, setTextVal] = useState<string>('');
 
-  const currentItem = useSharedValue<DrawItem | null>(null);
+    const currentItem = useSharedValue<DrawItem | null>(null);
 
-  const initialItem = useSharedValue<DrawItem | null>(null);
+    const initialItem = useSharedValue<DrawItem | null>(null);
 
-  const [doneItems, setDoneItems] = useState<DrawItem[]>([]);
+    const [doneItems, setDoneItems] = useState<DrawItem[]>([]);
```

Comment:
You should remove `doneItems` : `doneItems` should be the same as the last element of `screenStates` so you could use `screenStates.at(-1)` as `doneItems`